### PR TITLE
feat: add custom-copilot chat-with-data basic search python template

### PR DESCRIPTION
**Pull request by frankqianms**
workitem: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27479752

E2E TEST: N/A
Creating a new custom copilot RAG basic search bot app using python:
image
image
image
image
image

Conversation with bot:
image

### CODE REVIEW

File: templates/python/custom-copilot-rag-customize/teamsapp.local.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    with:
+      # Teams app name
+      name: {{appName}}${{TEAMSFX_ENV}}
+    # Write the information of created resources into environment file for
+    # the specified environment variable(s).
+    writeToEnvironmentFile: 
+      teamsAppId: TEAMS_APP_ID
+
+  # Create or reuse an existing Azure Active Directory application for bot.
+  - uses: botAadApp/create
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/teamsapp.local.yml.tpl:18]: Please use `aadApp/create` to create an AAD app for bot. You can refer other templates about how to do it.

Comment (blackchoey) [reply templates/python/custom-copilot-rag-customize/teamsapp.local.yml.tpl:18]: Please also update the yaml file for the azure ai search template.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -316,4 +316,5 @@
 /templates/python/* @adashen @blackchoey @frankqianms 
 /templates/python/custom-copilot-basic @frankqianms @blackchoey
 /templates/python/custom-copilot-rag-azure-ai-search @frankqianms @blackchoey
+/templates/python/custom-copilot-rag-customize @frankqianms @blackchoey
```
Comment (blackchoey) [inline .github/CODEOWNERS:319]: Since we already have line 316, maybe we don't need to add this line any more?

File: packages/fx-core/tests/question/create.test.ts [hunk 3]
```diff
@@ -1201,7 +1201,7 @@ describe("scaffold question", () => {
             const select = question as SingleSelectQuestion;
             const options = await select.dynamicOptions!(inputs);
             assert.isTrue(options.length === 3);
-            return ok({ type: "success", result: "typescript" });
+            return ok({ type: "success", result: "python" });
```
Comment (blackchoey) [inline packages/fx-core/tests/question/create.test.ts:1204]: Is this test case for typescript? I'm not sure whether changing to python will impact tests for typescript, can you help confirm with Yuan?

File: templates/python/custom-copilot-rag-customize/env/.env.local.user.tpl [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# This file includes environment variables that will not be committed to git by default. You can set these environment variables in your CI/CD system for your project.
+
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/env/.env.local.user.tpl:2]: Do we need to include `.env.local` and `.env.testtool` files in the template? I only see the `.user` file for them.

File: templates/python/custom-copilot-rag-customize/src/prompts/chat/config.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      "temperature": 0.9,
+      "top_p": 0.0,
+      "presence_penalty": 0.6,
+      "frequency_penalty": 0.0,
+      "stop_sequences": []
+  },
+  "augmentation": {
+      "augmentation_type": "none",
+      "data_sources": {
+        "local-ai-search": 1200
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/src/prompts/chat/config.json:20]: Do we want to give the data source another name? This template is not related to Azure AI search.

File: templates/python/custom-copilot-rag-customize/src/local_data_source.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+from teams.state.memory import Memory
+
+@dataclass
+class Result:
+    def __init__(self, output, length, too_long):
+        self.output = output
+        self.length = length
+        self.too_long = too_long
+
+class LocalAISearchDataSource(DataSource):
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/src/local_data_source.py:16]: Also update the class name if we don't want to include "AI Search" in the data source name.

File: templates/python/custom-copilot-rag-customize/env/.env.local.user.tpl (hunk 4 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/env/.env.local.user.tpl:2]: yes we do. I'll add them.

File: packages/fx-core/tests/question/create.test.ts (hunk 3 above)
Comment (frankqianms) [reply packages/fx-core/tests/question/create.test.ts:1204]: actually all languages here are the same. since ts code is not merged, we need to use python here

File: .github/CODEOWNERS (hunk 2 above)
Comment (frankqianms) [reply .github/CODEOWNERS:319]: ok, will remove it

File: templates/python/custom-copilot-rag-customize/src/local_data_source.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    """
+
+    def __init__(self, name):
+        """
+        Creates a new instance of the LocalAISearchDataSource instance.
+        """
+        self.name = name
+        self._data = []
+
+    def init(self):
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/src/local_data_source.py:28]: Are we able to consolidate the `init` function into `__init__`?

File: templates/python/custom-copilot-rag-customize/src/prompts/chat/config.json (hunk 5 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/src/prompts/chat/config.json:20]: sure, will rename it

File: templates/python/custom-copilot-rag-customize/src/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+python-dotenv
+aiohttp
+azure-search
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/src/requirements.txt:3]: do we need the azure-search related packages?

File: templates/python/custom-copilot-rag-customize/src/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+azure-search-documents
+teams-ai~=1.0.0
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/src/requirements.txt:5]: may need to change teams-ai version to 1.0.1?

File: templates/python/custom-copilot-rag-customize/src/requirements.txt (hunk 8 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/src/requirements.txt:3]: no we dont. will remove it

File: templates/python/custom-copilot-rag-customize/README.md.tpl [hunk 10]
```diff
@@ -0,0 +1,121 @@
+# Overview of the Basic RAG Bot template
+
+This template showcases a bot app that responds to user questions like an AI assistant according to data from Azure Search. This enables your users to talk with the AI assistant in Teams to find information.
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/README.md.tpl:3]: The data is not from Azure Search

File: templates/python/custom-copilot-rag-customize/teamsapp.yml.tpl [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    with:
+      # Teams app name
+      name: {{appName}}${{TEAMSFX_ENV}}
+    # Write the information of created resources into environment file for
+    # the specified environment variable(s).
+    writeToEnvironmentFile:
+      teamsAppId: TEAMS_APP_ID
+
+  # Create or reuse an existing Azure Active Directory application for bot.
+  - uses: botAadApp/create
```
Comment (blackchoey) [inline templates/python/custom-copilot-rag-customize/teamsapp.yml.tpl:21]: please also use `aadApp/create` here.

File: templates/python/custom-copilot-rag-customize/src/requirements.txt (hunk 9 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/src/requirements.txt]: updated

File: templates/python/custom-copilot-rag-customize/src/local_data_source.py (hunk 7 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/src/local_data_source.py]: done

File: templates/python/custom-copilot-rag-customize/src/local_data_source.py (hunk 6 above)
Comment (frankqianms) [reply templates/python/custom-copilot-rag-customize/src/local_data_source.py]: done

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — DISMISSED** [review]
LGTM.

**Review by jayzhang — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]
